src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -464,20 +470,31 @@ private void CreateOrUpdateDisabledSourcesBlock(XmlDocument nugetConfig, Diction
                 // If there's a clear node in the children of the disabledSources, we want to put any of our entries after the last one seen.
                 if (disabledSourcesNode.HasChildNodes)
                 {
-                    for (int i = 0; i < disabledSourcesNode.ChildNodes.Count; i++)
+                    XmlNode currentNode = disabledSourcesNode.FirstChild;
+
+                    while (currentNode != null)
                     {
-                        if (disabledSourcesNode.ChildNodes[i].Name.Equals("clear", StringComparison.InvariantCultureIgnoreCase))
+                        if (currentNode.Name.Equals("clear", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
I feel pretty confident about this change (has test coverage in literally every test, is a small change) but I did merge it this morning, is that a problem?

Reply:
so a green build should mean this change is tested?

Reply:
Yes; I made sure that without the fixes the tests failed (by adding the double-call) and that no expected inputs or outputs had to be changed.  It's just that I checked in recently, so I'll let you make that call.

Reply:
Perfect, then there should be no issues in taking this change.